integration_test/cases/preload.exs

```diff
@@ -708,6 +709,82 @@ defmodule Ecto.Integration.PreloadTest do
     assert %User{id: ^uid1} = item.user
   end
 
+  describe "Preload associations in embedded_schema from container schema" do
+    setup do
+      # Order embeds_many items
+      # Items belong to Users
+      # Preload Users from an Order schema
+      %User{id: uid1} = TestRepo.insert!(%User{name: "1"})
+      %User{id: uid2} = TestRepo.insert!(%User{name: "2"})
+      %User{id: uid3} = TestRepo.insert!(%User{name: "3"})
+      item1 = %Item{id: 1, user_id: uid1}
+      item2 = %Item{id: 2, user_id: uid2}
+      item3 = %Item{id: 3, user_id: uid3}
+      order1 = %Order{items: [item1, item3, item2],
+                      item: item1}
+      order2 = %Order{items: [item1, item2],
+                      item: item2}
+      order3 = %Order{items: [item3, item2, item1],
+                      item: item3}
+      order4 = %Order{items: [],
+                      item: item3}
+      TestRepo.insert!(order1)
+      TestRepo.insert!(order2)
+      TestRepo.insert!(order3)
+
+      Enum.map([order1, order2, order3], fn order ->
```

Comment:
```suggestion
      order4 = %Order{items: [],
                      item: nil}
```
We don't need to insert the orders because the preload will just take the user ids and put them into these structs, even if the orders are not in the db.

Also we should have a test case where `embed_one` is `nil`.

Reply:
I tried to search the docs for the possibility of `embed_one` being `nil` but did not find much, I also did not search too much. Adding it to the tests 👍